File: retroauth/src/test/java/com/andretietz/retroauth/testimpl/TestTokenStorage.java
package com.andretietz.retroauth.testimpl;

import com.andretietz.retroauth.TokenStorage;

public class TestTokenStorage implements TokenStorage<String, String, String> {

    public static final String TEST_TOKEN = "token";
    private TestBehaviour behaviour;

    @Override
    public String getToken(String owner, String tokenType) {
        if (behaviour != null) {
            return behaviour.getToken(owner, tokenType);
        }
        return TEST_TOKEN;
    }

    @Override
    public void removeToken(String s, String s2, String s3) {

    }

    @Override
    public void storeToken(String s, String s2, String s3) {

    }

    public void setTestBehaviour(TestBehaviour behaviour) {
        this.behaviour = behaviour;
    }

    public interface TestBehaviour {
        String getToken(String owner, String tokenType);
    }
}

File: retroauth/src/test/java/com/andretietz/retroauth/testimpl/TestTokenStorage.kt
package com.andretietz.retroauth.testimpl

import com.andretietz.retroauth.TokenStorage

open class TestTokenStorage : TokenStorage<String, String, String> {

    companion object {
        const val TEST_TOKEN = "token"
    }

    private var tmpToken: String? = null


    override fun getToken(owner: String, type: String): String {
        return tmpToken ?: TEST_TOKEN
    }

    override fun removeToken(owner: String, type: String, token: String) {

    }

    override fun storeToken(owner: String, type: String, token: String): String {
        tmpToken = token
        return token
    }
}
